x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/Function.java

```diff
@@ -75,4 +77,31 @@ protected String functionArgs() {
     public boolean functionEquals(Function f) {
         return f != null && getClass() == f.getClass() && arguments().equals(f.arguments());
     }
+
+    /**
+     * Build the string representation of the function arguments for the {@link #toString()}.
+     *  - If an argument is a Function call {@code toString()} to work recursively
+     *  - If an argument is a Literal then just use the evaluated value.
+     *  - If an argument is a NamedExpression then use it's {@code name()} to avoid long confusing output
+     *  - If none of those resolve to the {@code toString()} method of the argument
+     *
+     * @return String representation of the function args for the {@link #toString} representation
+     */
+    private String functionArgsForToString() {
+        StringJoiner sj = new StringJoiner(",", "(", ")");
+        for (Expression child : children()) {
+            String val;
+            if (child instanceof Function) {
```

Comment:
This looks like a good case of overloading/method dispatch.
Literal already returns its value or, in case it is named, name=value.
Function does something similar I believe (especially for operators).